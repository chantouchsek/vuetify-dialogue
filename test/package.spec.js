import Vue from 'vue'
import Vuetify from 'vuetify/lib'
import VuetifyDialogPlugin from '../src'
import { sleep, addElemWithDataAppToBody, disableTransitions } from './utils'
Vue.use(Vuetify)
const vuetify = new Vuetify({})

describe('manager', () => {
  disableTransitions()
  addElemWithDataAppToBody()
  Vue.use(VuetifyDialogPlugin, {
    confirm: {
      actions: ['Foo', 'Bar', 'Baz'],
      icon: false
    },
    context: {
      vuetify
    }
  })
  const manager = Vue.prototype.$dialog

  test('Test confirmation with generic options', async () => {
    const dialog = await manager.confirm({
      waitForResult: false,
      text: 'Test confirmation with generic options',
      title: 'Title'
    })
    await sleep(100)
    expect(dialog.element).toMatchSnapshot()
    await dialog.close()
    await Vue.nextTick()
    await sleep(5)
    expect(document.body).toMatchSnapshot()
  })
})
