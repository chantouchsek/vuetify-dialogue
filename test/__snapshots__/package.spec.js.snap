// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manager Test confirmation with generic options 1`] = `
<div
  class="v-dialog__container"
  role="dialog"
/>
`;

exports[`manager Test confirmation with generic options 2`] = `
<body>
  <div
    data-app="true"
  >
    <div
      class="v-overlay theme--dark"
      style="z-index: 201;"
    >
      <div
        class="v-overlay__scrim"
        style="opacity: 0; background-color: rgb(33, 33, 33); border-color: #212121;"
      />
    </div>
    <div
      class="v-dialog__content"
      role="document"
      style="z-index: 202; z-index: 202;"
    >
      <div
        class="v-dialog vuedl-layout dialog-transition-leave dialog-transition-leave-active"
        style="transform-origin: center center; max-width: 450px;"
      >
        <div
          class="v-dialog-wrapper"
        >
          <div
            class="vuedl-layout__closeBtn"
          >
            
      Ã—
    
          </div>
           
          <div
            class="v-card v-sheet theme--light rounded-0"
          >
            <header
              class="v-sheet theme--light v-toolbar v-toolbar--dense v-toolbar--flat"
              style="height: 48px;"
            >
              <div
                class="v-toolbar__content"
                style="height: 48px;"
              >
                <!---->
                 
                <div
                  class="v-toolbar__title"
                >
                  
      Title
    
                </div>
              </div>
            </header>
             
            <div
              class="v-card__text body-1 py-2"
            >
              Test confirmation with generic options
            </div>
             
            <div
              class="v-card__actions"
            >
              <div
                class="spacer"
              />
               
              <button
                action-key="Foo"
                class="v-btn v-btn--flat v-btn--text theme--light v-size--default"
                type="button"
              >
                <span
                  class="v-btn__content"
                >
                  <!---->
                  
  Foo
  
                  <!---->
                </span>
              </button>
              <button
                action-key="Bar"
                class="v-btn v-btn--flat v-btn--text theme--light v-size--default"
                type="button"
              >
                <span
                  class="v-btn__content"
                >
                  <!---->
                  
  Bar
  
                  <!---->
                </span>
              </button>
              <button
                action-key="Baz"
                class="v-btn v-btn--flat v-btn--text theme--light v-size--default"
                type="button"
              >
                <span
                  class="v-btn__content"
                >
                  <!---->
                  
  Baz
  
                  <!---->
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="v-dialog__container"
      role="dialog"
    />
  </div>
</body>
`;
